zerver/tornado/views.py

```diff
@@ -47,10 +39,8 @@ async def wrapped(*args: P.args, **kwargs: P.kwargs) -> T:
 
 
 @internal_api_view(True)
-@has_request_variables
-def notify(
-    request: HttpRequest, data: Mapping[str, Any] = REQ(json_validator=check_dict([]))
-) -> HttpResponse:
+@typed_endpoint
+def notify(request: HttpRequest, *, data: Json[dict[str, Any]]) -> HttpResponse:
```

Comment:
It would be a good follow-up refinement to add a more precise type that `process_notification` expects here.